lib/book.py

```diff
@@ -244,45 +148,41 @@ def _search_shelfari(self, connection, keywords):
             try:
                 connection.close()
                 if self._proxy:
-                    connection = HTTPConnection(self._http_address, self._http_port)
-                    connection.set_tunnel('www.shelfari.com', 80)
+                    connection = HTTPSConnection(self._https_address, self._https_port)
+                    connection.set_tunnel('www.goodreads.com', 443)
                 else:
-                    connection = HTTPConnection('www.shelfari.com')
+                    connection = HTTPSConnection('www.goodreads.com')
 
                 connection.request('GET', '/search/books?' + query)
                 response = connection.getresponse().read()
             except:
                 self._status = self.FAIL
-                self._status_message = self.FAILED_COULD_NOT_CONNECT_TO_SHELFARI
+                self._status_message = self.FAILED_COULD_NOT_CONNECT_TO_GOODREADS
                 raise Exception(self._status_message)
         
         # check to make sure there are results
         if 'did not return any results' in response:
             return connection
-        urlsearch = self.SHELFARI_URL_PAT.search(response)
+        urlsearch = self.GOODREADS_URL_PAT.search(response)
```

Comment:
This is a bug - can't work, as it isn't defined (it is in BookSettings).  However, I can't find a way to hit this codepath - even if I clear the goodreads URL in the per-book JSON file, connect a kindle, and do "send to device" (the only path that could come through here) it still doesn't fail.  Looks like the goodreads URL is being found elsewhere.

Of course, if the goodreads lookup failed then we'd hit problems here, but we might have already barfed earlier..

Is this a redundant code path?